buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_26_0.groovy

```diff
@@ -0,0 +1,197 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.gradle
+
+import org.gradle.api.Project
+
+/**
+ * Utilities for working with our vendored version of gRPC. The test dependency junit and the runtime dependencies
+ * slf4j, commons-logging and log4j are not included in the vendored jar. The linkage checker complains about
+ * these dependencies could be ignored.
+ */
+class GrpcVendoring_1_26_0 {
+  /** Returns the list of compile time dependencies. */
+  static List<String> dependencies() {
+    return [
+      'com.google.guava:guava:26.0-jre',
+      'com.google.protobuf:protobuf-java:3.11.0',
+      'com.google.protobuf:protobuf-java-util:3.11.0',
+      'com.google.code.gson:gson:2.8.6',
+      'io.grpc:grpc-auth:1.26.0',
+      'io.grpc:grpc-core:1.26.0',
+      'io.grpc:grpc-context:1.26.0',
+      'io.grpc:grpc-netty:1.26.0',
+      'io.grpc:grpc-protobuf:1.26.0',
+      'io.grpc:grpc-stub:1.26.0',
+      'io.netty:netty-transport-native-epoll:4.1.42.Final',
+      // tcnative version from https://github.com/grpc/grpc-java/blob/master/SECURITY.md#netty
+      'io.netty:netty-tcnative-boringssl-static:2.0.26.Final',
+      'com.google.auth:google-auth-library-credentials:0.18.0',
+      'io.grpc:grpc-testing:1.26.0',
+      'com.google.api.grpc:proto-google-common-protos:1.12.0',
+      'io.opencensus:opencensus-api:0.24.0',
+      'io.opencensus:opencensus-contrib-grpc-metrics:0.24.0',
+      'io.perfmark:perfmark-api:0.19.0',
+      'com.github.jponge:lzma-java:1.3',
+      'com.google.protobuf.nano:protobuf-javanano:3.0.0-alpha-5',
+      'com.jcraft:jzlib:1.1.3',
+      'com.ning:compress-lzf:1.0.3',
+      'net.jpountz.lz4:lz4:1.3.0',
+      'org.bouncycastle:bcpkix-jdk15on:1.54',
+      'org.bouncycastle:bcprov-jdk15on:1.54',
+      'org.conscrypt:conscrypt-openjdk-uber:1.3.0',
+      'org.eclipse.jetty.alpn:alpn-api:1.1.2.v20150522',
+      'org.eclipse.jetty.npn:npn-api:1.1.1.v20141010',
+      'org.jboss.marshalling:jboss-marshalling:1.4.11.Final',
+      'org.jboss.modules:jboss-modules:1.1.0.Beta1'
+    ]
+  }
+
+  /**
+   * Returns the list of runtime time dependencies that should be exported as runtime
+   * dependencies within the vendored jar.
+   */
+  static List<String> runtimeDependencies() {
+    return [
+      'com.google.errorprone:error_prone_annotations:2.3.3',
+      'commons-logging:commons-logging:1.2',
+      'org.apache.logging.log4j:log4j-api:2.6.2',
+      'org.slf4j:slf4j-api:1.7.21'
+    ]
+  }
+
+  /**
+   * Returns the list of test dependencies.
+   */
+  static List<String> testDependencies() {
+    return [
+      'junit:junit:4.12',
```

Comment:
We should make this a provided dependency, and not test since test will only be part of the test runtime for the specific pom while we want users who are consuming this dependency to provide junit if they intend to test using the vendored grpc library.